cpp/src/DataStorm/Contract.ice

```diff
@@ -391,6 +391,31 @@ module DataStormContract
         void s(long topicId, long elementId, DataSample sample);
     }
 
+    /// The reason for the session creation failure.
+    enum SessionError
+    {
+        // The session is already connected.
+        AlreadyConnected,
+
+        // Node is shutting down.
+        NodeShutdown,
+
+        // The connection failed
+        ConnectionFailure,
+
+        // A confirmation was received for a session that doesn't exist.
+        SessionNotFound,
+
+        // The session failed for an unknown reason.
+        UnknownError,
```

Comment:
Maybe just `Unknown`. We don't prepend Error to anything else.

Reply:
Or Internal[Error], like StatusCode::InternalError in IceRPC. Means "some other error without a specific code for the caller".